provider/provider.go

```diff
@@ -117,20 +118,38 @@ func (p *dockerNativeProvider) Check(ctx context.Context, req *rpc.CheckRequest)
 
 	dockerContext := build.Context
 	dockerfile := build.Dockerfile
-
-	// Set docker build context
+	// Hash docker build context digest
 	contextDigest, err := hashContext(dockerContext, dockerfile)
 	if err != nil {
 		return nil, err
 	}
 
-	// add implicit resource to provider
+	// Get the system platform
+	os := runtime.GOOS
+	// Docker for Mac builds and runs for linux/amd64, not darwin
+	if os == "darwin" {
+		os = "linux"
+	}
+	arch := runtime.GOARCH
+	hostPlatform := filepath.Join(os, arch)
+	msg := fmt.Sprintf(
+		"Building your image for %s architecture.\n"+
+			"To ensure you are building for the correct platform, consider "+
+			"explicitly setting the `platform` field on ImageBuildOptions.", hostPlatform)
+
+	// build options: add implicit resource contextDigest and set default host platform
 	if inputs["build"].IsNull() {
 		inputs["build"] = resource.NewObjectProperty(resource.PropertyMap{
 			"contextDigest": resource.NewStringProperty(contextDigest),
+			"platform":      resource.NewStringProperty(hostPlatform),
 		})
+		p.host.Log(ctx, "info", urn, msg)
 	} else {
 		inputs["build"].ObjectValue()["contextDigest"] = resource.NewStringProperty(contextDigest)
+		if inputs["build"].ObjectValue()["platform"].IsNull() {
```

Comment:
Slight worry what if inputs["build"] is not an object and this would panic. But that's probably OK for now. What I'd love to see instead is facilities to bind PropertyValue to typed Go models so code like this can operate on typed Go structs instead of on PropertyValue and be less error prone. But that's for later.

Reply:
Ah! No worries for now - there's a null check for `inputs["build"] on line 141.
I see your overall point, that would be a good refactor given how much we're using PropertyValue to do logic here.

Reply:
Well if it's string and not an object it can still panic. But its' too much defensive coding to put that check in.

Reply:
right! we have to close the loop on https://github.com/pulumi/pulumi-docker/issues/428 here.